firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -753,7 +770,7 @@ private Type resolveType(Type type, Map<TypeVariable<Class<T>>, Type> types) {
       }
     }
 
-    Map<String, Object> serialize(T object, ErrorPath path) {
+    Map<String, Object> serialize(T object, ErrorPath path, CustomClassMapper classMapper) {
```

Comment:
Given that the `BeanMapper` is shared between instances of `CustomClassMapper` and that we already need to pass a reference to `CustomClassMapper` around, would in make sense to just leave everything static and instead have a `SerializationContext` that included both the document-id and the `ErrorPath`?

It doesn't seem like that would blow the change up that much but would at least keep the way the document-id is accessed uniform.

Reply:
Made it back to all static with DeserializeContext, since we don't need the doc id for Serialization any more.